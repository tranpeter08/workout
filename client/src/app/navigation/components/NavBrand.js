import React from 'react';
import {NavLink} from 'react-router-dom';
import logo from '../strong-white.png';

export default function NavBrand({closeNav}) {
  return (
    <NavLink to='/' className='nav-brand' onClick={closeNav}>
      <h1 className='nav-brand-font'>GET SWOLE</h1>
      <img className='nav-brand-img' src={logo} alt='Logo of strong arm'/>
    </NavLink>
  )
}