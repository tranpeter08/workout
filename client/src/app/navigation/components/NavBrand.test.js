import React from 'react';
import NavBrand from './NavBrand';
import {shallow, mount, render} from 'enzyme';
import {BrowserRouter as Router} from 'react-router-dom';

describe('<NavBrand />', () => {
  let wrapper;
  it('should render without crahsing', () => {
    wrapper = shallow(<NavBrand />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain the right heading', () => {
    const wrapper = shallow(<NavBrand />);
    expect(wrapper.find('.nav-brand-font').text()).toEqual('GET SWOLE');
  });

  it('should call closeNav when clicked', () => {
    const closeNav = jest.fn();
    const component = (
      <NavBrand closeNav={closeNav} />
    );

    wrapper = shallow(component);
    wrapper
      .find('.nav-brand')
      .simulate('click');
      expect(closeNav).toHaveBeenCalled();
  });
});