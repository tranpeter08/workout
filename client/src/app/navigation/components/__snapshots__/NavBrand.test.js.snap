// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBrand /> should render without crahsing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBrand />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "aria-current": "page",
      "children": Array [
        <h1
          className="nav-brand-font"
        >
          GET SWOLE
        </h1>,
        <img
          alt="Logo of strong arm"
          className="nav-brand-img"
          src="strong-white.png"
        />,
      ],
      "className": "nav-brand",
      "onClick": undefined,
      "to": "/",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "GET SWOLE",
          "className": "nav-brand-font",
        },
        "ref": null,
        "rendered": "GET SWOLE",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Logo of strong arm",
          "className": "nav-brand-img",
          "src": "strong-white.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": Array [
          <h1
            className="nav-brand-font"
          >
            GET SWOLE
          </h1>,
          <img
            alt="Logo of strong arm"
            className="nav-brand-img"
            src="strong-white.png"
          />,
        ],
        "className": "nav-brand",
        "onClick": undefined,
        "to": "/",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "GET SWOLE",
            "className": "nav-brand-font",
          },
          "ref": null,
          "rendered": "GET SWOLE",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Logo of strong arm",
            "className": "nav-brand-img",
            "src": "strong-white.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
